### ISSUE TITLE
Broken advection after geometric advection. -- Or maybe broken boolean operation?

### ISSUE THREAD

**Issue by AdmiralCurtiss** [ViennaTools/ViennaLS#64]
This one is very confusing to me. Basically, if we run a regular growing `lsAdvect` after a `lsGeometricAdvect`, the growth behaves very strangely, as if part of the mask still existed in some form inside the levelset.

![image](https://user-images.githubusercontent.com/4522237/128453623-2a4fa3d5-2035-4c38-8ef3-657f481cfa7e.png)

Orange is the levelset as it is after the geometric advection. White is the result after applying what should be a simple isotropic deposition on it.

Here's sample code that replicates the above image: https://gist.github.com/AdmiralCurtiss/4218aa814158ecad23b2f77e5e858a9e

Are we doing anything wrong here, or is there an actual bug somewhere?

**Comment by AdmiralCurtiss** [comment]
Second example that I ran across while trying to replicate something else: https://gist.github.com/AdmiralCurtiss/4a9820635c30d5eaeaf51434b634860b

**Comment by XaverKlemenschits** [comment]
The behaviour you observed was actually due to some exact zeros that were not coped with properly. I have just merged the fix for these problems.

However, the strategy you use to generate the initial geometry is unnecessarily complicated. I made some sample code to show you how it can be done more easily, which also avoids you having to set your own points in a mesh by hand:

https://gist.github.com/XaverKlemenschits/4bb9e9d202d94ae832a4c5ff0c727c94
